stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -293,15 +293,24 @@ true
 @inline mul!(C::AbstractMatrix, A::AbstractVecOrMat, B::AbstractVecOrMat, α::Number, β::Number) = _mul!(C, A, B, α, β)
 # Add a level of indirection and specialize _mul! to avoid ambiguities in mul!
 @inline _mul!(C::AbstractMatrix, A::AbstractVecOrMat, B::AbstractVecOrMat, α::Number, β::Number) =
-    generic_matmatmul!(
+    generic_matmatmul_wrapper!(
         C,
         wrapper_char(A),
         wrapper_char(B),
         _unwrap(A),
         _unwrap(B),
-        α, β
+        α, β,
+        Val(isNTC(A) & isNTC(B))
```

Comment:
At this point, the wrappers are not unwrapped yet, so this could be simply

```julia
isNTC(::AbstractArray) = true # 'N'
isNTC(::Adjoint) = true # 'C'
isNTC(::Tranpose) = true # 'T'
isNTC(::Symmetric) = false # 'S' or 's'
isNTC(::Hermitian) = false # 'H' or 'h'
```

Reply:
I didn't go this route to protect against new wrappers defining their own characters. Perhaps we can do this for the specific wrapper types, but with a check for `AbstractArrays`, i.e., `isNTC(A::AbstractArray) = uppercase(wrapperchar(A)) == 'N'`?

Reply:
Sounds good.